### PR TITLE
Cherry pick testing commits from confluence to streamflow

### PR DESCRIPTION
**Pull request by yondonfu**
**What does this pull request do? Explain your changes. (required)**

This PR cherry picks a few commits with a few test suite fixes from the `confluence` branch that are useful for the `streamflow` branch as well.

Update:

- Also added prettier-eslint, ran prettier on all js/ts/sol files and updated the Husky pre-commit hook to run prettier prior to linting

**Specific updates (required)**

See commit history. Refer to the `confluence` [commit history](https://github.com/livepeer/protocol/commits/confluence) for the cherry picked commits.

**How did you test each of these updates (required)**

Ran test suite.

**Does this pull request close any open issues?**

N/A

**Checklist:**

- [x] README and other documentation updated
- [x] All tests using `yarn test` pass

### CODE REVIEW

File: test/integration/PoolUpdatesWithHints.js [hunk 1]
```diff
@@ -121,56 +135,100 @@ describe("PoolUpdatesWithHints", () => {
...
         const txResNoHint = await tx.wait()
+        await expect(tx.hash)
+            .to.emit(bondingManager, "TranscoderDeactivated")
+            .withArgs(transcoders[size - 1].address, dr)
 
         assert.equal(await transcoderAtPoolPos(size - 1), newTranscoder.address)
-        expect(tx).to.emit(bondingManager, "TranscoderDeactivated").withArgs(transcoders[size - 1].address, dr)
+        await expect(tx)
+            .to.emit(bondingManager, "TranscoderDeactivated")
+            .withArgs(transcoders[size - 1].address, dr)
```
Comment (kautukkundan) [inline test/integration/PoolUpdatesWithHints.js:201]: Can actually remove this
Lines 197-199 do the same thing.

File: test/integration/PoolUpdatesWithHints.js [hunk 2]
```diff
@@ -121,56 +135,100 @@ describe("PoolUpdatesWithHints", () => {
...
+        await expect(tx.hash)
+            .to.emit(bondingManager, "TranscoderDeactivated")
+            .withArgs(transcoders[size - 1].address, dr)
+
         const txResHint = await tx.wait()
         assert.equal(await transcoderAtPoolPos(size - 1), newTranscoder.address)
-        expect(tx).to.emit(bondingManager, "TranscoderDeactivated").withArgs(transcoders[size - 1].address, dr)
+        await expect(tx)
+            .to.emit(bondingManager, "TranscoderDeactivated")
+            .withArgs(transcoders[size - 1].address, dr)
```
Comment (kautukkundan) [inline test/integration/PoolUpdatesWithHints.js:219]: Can remove this as well
Lines 217-219 do the same thing.

**Review by kautukkundan — DISMISSED** [review]
LGTM 👍🏼 
Minor changes

File: test/integration/PoolUpdatesWithHints.js (hunk 2 above)
Comment (yondonfu) [reply test/integration/PoolUpdatesWithHints.js:219]: Fixed in 6ac1cb5

File: test/integration/PoolUpdatesWithHints.js (hunk 1 above)
Comment (yondonfu) [reply test/integration/PoolUpdatesWithHints.js:201]: Fixed in 6ac1cb5

**Pull request merged**